### PR TITLE
[GlobalISel][AArch64] Legalize G_UADDSAT, G_SADDSAT, G_USUBSAT, and G…

### PR DESCRIPTION
**Pull request by tschuett**
…_SSUBSAT

sve-int-imm.ll also tests saturation, but it has unsupported splats.

### CODE REVIEW

File: llvm/test/CodeGen/AArch64/sve-saturating-arith.ll [hunk 1]
```diff
@@ -0,0 +1,171 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -mtriple=aarch64 -mattr=+sve -verify-machineinstrs %s -o - | FileCheck %s --check-prefixes=CHECK,CHECK-SDAG
```
Comment (madhur13490) [inline llvm/test/CodeGen/AArch64/sve-saturating-arith.ll:2]: CHECK-SDAG is unused. remove it?

File: llvm/test/CodeGen/AArch64/sve-saturating-arith.ll [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+; RUN: llc -mtriple=aarch64 -mattr=+sve -aarch64-enable-gisel-sve=1 -global-isel -global-isel-abort=2 -verify-machineinstrs %s -o - 2>&1 | FileCheck %s --check-prefixes=CHECK,CHECK-GISEL
```
Comment (madhur13490) [inline llvm/test/CodeGen/AArch64/sve-saturating-arith.ll:3]: Two check prefixes not needed

Comment (tschuett) [reply llvm/test/CodeGen/AArch64/sve-saturating-arith.ll]: That was blind copy-and-paste.

File: llvm/lib/CodeGen/GlobalISel/LegalityPredicates.cpp [hunk 3]
```diff
@@ -204,6 +204,8 @@ LegalityPredicate LegalityPredicates::memSizeInBytesNotPow2(unsigned MMOIdx) {
 LegalityPredicate LegalityPredicates::memSizeNotByteSizePow2(unsigned MMOIdx) {
   return [=](const LegalityQuery &Query) {
     const LLT MemTy = Query.MMODescrs[MMOIdx].MemoryTy;
+    if (Query.MMODescrs[MMOIdx].MemoryTy.isScalableVector())
+      return true;
```
Comment (davemgreen) [inline llvm/lib/CodeGen/GlobalISel/LegalityPredicates.cpp:208]: Is this needed for addsat?

**Review by davemgreen — COMMENTED** [review]
This sounds OK, but is the memsize part needed for this patch?

File: llvm/lib/CodeGen/GlobalISel/LegalityPredicates.cpp (hunk 3 above)
Comment (tschuett) [reply llvm/lib/CodeGen/GlobalISel/LegalityPredicates.cpp]: Nope. Probably left over from debugging.

**Review by davemgreen — APPROVED** [review]
Thanks. LGTM

**Pull request merged**